django/db/migrations/operations/models.py

```diff
@@ -122,6 +122,10 @@ def state_forwards(self, app_label, state):
         del state.models[app_label, self.old_name.lower()]
         # Repoint the FKs and M2Ms pointing to us
         for related_object in (related_objects + related_m2m_objects):
+            if related_object.parent_model is not model:
```

Comment:
I think we should go with `related_object.field.rel.to is not model` instead. It's more self explaining and should do the same.